app/experimenter/nimbus-ui/README.md

```diff
@@ -197,6 +166,69 @@ if (message !== "success") {
 console.log("Retrieved record", record);
 ```
 
+## Forms and Validation
+
+**See the form validation section of [Storybook](https://storage.googleapis.com/mozilla-storybooks-experimenter/index.html) for examples and a visual guide to the UX interaction.**
+
+There are three kinds of validation that need to be applied to forms: client-side, server-side, and "required for launch".
+
+### `useCommonForm` and `useCommonNestedForm`
+
+All forms use [`react-bootstrap`](https://react-bootstrap.github.io/components/forms/) as well as the custom convenience hooks `useCommonForm` or `useCommonNestedForm` for ease of use when developing new forms, to promote consistent behavior amongst our forms, and to limit any future changes affecting all forms in limited places. These hooks utilize the [`react-hook-form`](https://react-hook-form.com/api) package.
+
+If there are no nested forms, pass the needed parameters into `useCommonForm`. Otherwise, pull form methods out of _our_ `useForm` hook and pass them into `FormContext`. Then, in the nested form(s), pull out the methods and pass them into `useCommonNestedForm`.
```

Comment:
A link to the source code would be nice from `needed parameters`

Reply:
I'm a little hesitant since I think it's very possible that it'll become out of date later, we have a TODO a the top of the file and we'll also need some edits in that file when I have time to address #4485 😬